### ISSUE TITLE
jupyter lab --generate-config creates a file that is never read

### ISSUE THREAD

**Issue by ariel-co** [jupyterlab/jupyterlab#10325]
## Description

The `jupyter lab --generate-config` command creats a file that `jupyter lab` never looks for.

## Reproduce

```
> jupyter lab --generate-config
Writing default config to: C:\Users\ariel\.jupyter\jupyter_lab_config.py
```

## Expected behavior

Not sure if the output filename should be different or if ServerApp should look for `jupyter_lab_config`. Renaming it to `jupyter_lab.py` allows ServerApp to see it.

## Context

- Operating System and version: Windows 10 Pro
- JupyterLab version: 
```
  jupyter_client     conda-forge/noarch::jupyter_client-6.1.12-pyhd8ed1ab_0
  jupyter_core       conda-forge/win-64::jupyter_core-4.7.1-py39hcbf5309_0
  jupyter_server     conda-forge/noarch::jupyter_server-1.8.0-pyhd8ed1ab_0
  jupyterlab         conda-forge/noarch::jupyterlab-3.0.16-pyhd8ed1ab_0
  jupyterlab_pygmen~ conda-forge/noarch::jupyterlab_pygments-0.1.2-pyh9f0ad1d_0
  jupyterlab_server  conda-forge/noarch::jupyterlab_server-2.6.0-pyhd8ed1ab_0
```

<details><summary>Troubleshoot Output</summary>
<pre>
(jlab-test) PS C:\Users\ariel> jupyter lab --debug --no-browser
[D 2021-06-01 23:42:36.368 ServerApp] Searching ['C:\\Users\\ariel', 'C:\\Users\\ariel\\.jupyter', 'C:\\Users\\ariel\\miniconda3\\envs\\jlab-test\\etc\\jupyter', 'C:\\ProgramData\\jupyter'] for config files
[D 2021-06-01 23:42:36.368 ServerApp] Looking for jupyter_config in C:\ProgramData\jupyter
[D 2021-06-01 23:42:36.369 ServerApp] Looking for jupyter_config in C:\Users\ariel\miniconda3\envs\jlab-test\etc\jupyter
[D 2021-06-01 23:42:36.369 ServerApp] Looking for jupyter_config in C:\Users\ariel\.jupyter
[D 2021-06-01 23:42:36.370 ServerApp] Loaded config file: C:\Users\ariel\.jupyter\jupyter_config.py
[D 2021-06-01 23:42:36.371 ServerApp] Looking for jupyter_config in C:\Users\ariel
[D 2021-06-01 23:42:36.373 ServerApp] Looking for jupyter_server_config in C:\ProgramData\jupyter
[D 2021-06-01 23:42:36.373 ServerApp] Looking for jupyter_server_config in C:\Users\ariel\miniconda3\envs\jlab-test\etc\jupyter
[D 2021-06-01 23:42:36.374 ServerApp] Looking for jupyter_server_config in C:\Users\ariel\.jupyter
[D 2021-06-01 23:42:36.374 ServerApp] Looking for jupyter_server_config in C:\Users\ariel
[D 2021-06-01 23:42:36.377 ServerApp] Paths used for configuration of jupyter_server_config:
        C:\ProgramData\jupyter\jupyter_server_config.json
[D 2021-06-01 23:42:36.378 ServerApp] Paths used for configuration of jupyter_server_config:
        C:\Users\ariel\miniconda3\envs\jlab-test\etc\jupyter\jupyter_server_config.d\jupyterlab.json
        C:\Users\ariel\miniconda3\envs\jlab-test\etc\jupyter\jupyter_server_config.d\nbclassic.json
        C:\Users\ariel\miniconda3\envs\jlab-test\etc\jupyter\jupyter_server_config.json
[D 2021-06-01 23:42:36.380 ServerApp] Paths used for configuration of jupyter_server_config:
        C:\Users\ariel\.jupyter\jupyter_server_config.json</pre>
</details>

**Comment by spencermathews** [comment]
Yes, some clarity on this would be nice. I'm finding that, without any user config files (i.e. no `jupyter_lab_config.py` or `jupyter_server_config.py`), `jupyter lab` spawns a browser (despite indications that `open_browser` should default to `False`), yet `jupyter server` does not.

Uncommenting `c.ExtensionApp.open_browser = False` in `jupyter_lab_config.py` or `c.ServerApp.open_browser = False` in `jupyter_server_config.py` does prevent `jupyter lab` from opening a browser. As an aside, there are three ways to set `open_browser` in `jupyter_lab_config.py`, which is especially confusing since it's not even mentioned in the [JupyterLab Documentation](https://jupyterlab.readthedocs.io/).

Indeed, running `jupyter lab` using `--debug` (as you did) or `--show-config` there is no explicit mention of `jupyter_lab_config.py` being read, but I see config values set there clearly showing up in the output.

I can't tell what config options you are trying to set, but perhaps test them (and try `--show-config`) to see if they actually do take effect. It appears that the file is actually being read, but there is a bug in the `--debug` output that prevents it from being logged. I can't comment on your mention of file renaming since there is no output to accompany it.

My best guess is that the docs haven't caught up with [Standalone Jupyter server enhancement](https://jupyter.org/enhancement-proposals/28-jupyter-server/jupyter-server.html) JEP, and the best practice is to set config in `jupyter_server_config.py` if possible, unless your options are JupyterLab specific.
